CONTRIBUTING.md

```diff
@@ -77,14 +77,65 @@ For each function, its files spread in the follow places:
   should follow the [template][example-template].
 - The `tests/` directory contains additional e2e tests.
 
-### E2E Tests
+For golang-based functions, you need to generate some doc related variables from
+the `README.md` by running
+
+```shell
+$ cd functions/go
+$ make generate
+```
+
+### Tests
+
+#### Unit Tests
+
+To run all unit tests
+
+```shell
+$ make unit-test
+```
+
+#### E2E Tests
 
 The e2e tests are the recommended way to test functions in the catalog. They are
-very easy to write and set up with our [e2e test harness].
+very easy to write and set up with our e2e test harness. You can find all the
+supported options [here][e2e test harness doc].
 
 You can choose to put the e2e test in either the `examples/` directory or in the
 `tests/` directory depending on if it is worthwhile to be shown as an example.
 
+To test all examples and e2e tests, run
+
+```shell
+$ make e2e-test
+```
```

Comment:
I think most folks may not need this. IIRC this took +15m

Reply:
Updated